lib/ZwaveClient.ts

```diff
@@ -140,17 +140,61 @@ const allowedApis = validateMethods([
 	'abortInclusion',
 ] as const)
 
+export type CmdClsExistsMapping = {
+	commandClass: CommandClasses
+	existsValue: any
+	absentValue: any
+}
+
+export enum NodePropMappingAggFn {
+	ALL = 'all',
+	MIN = 'min',
+	MAX = 'max',
+	FIRST = 'first',
+	LAST = 'last',
+}
+
+export type NodePropMapping = {
+	hasCommandClass?: CmdClsExistsMapping
+	valueId?: ValueID
+	aggregationType?: NodePropMappingAggFn
+}
+
 // Define mapping of node properties to ValueIDs
-const mappedNodeProps: Record<string, ValueID> = {
-	batteryLevel: {
-		commandClass: CommandClasses.Battery,
-		property: 'level',
+const mappedNodeProps: Record<string, NodePropMapping> = {
+	batteryLevelMin: {
+		valueId: {
+			commandClass: CommandClasses.Battery,
+			property: 'level',
+		},
+		aggregationType: NodePropMappingAggFn.MIN,
+	},
+	batteryLevels: {
+		valueId: {
+			commandClass: CommandClasses.Battery,
+			property: 'level',
+		},
+		aggregationType: NodePropMappingAggFn.ALL,
+	},
+	powerSource: {
+		hasCommandClass: {
+			commandClass: CommandClasses.Battery,
+			existsValue: 'battery',
+			absentValue: 'mains',
+		},
```

Comment:
I think this could be made a bit cleaner, this could become a record of CC -> propMap like:

```js
const mappedNodeProps = {
[CommandClasses.Battery]: [
{
	nodeProp: 'batteryLevelMin',
	fn: (node, value) => {
		// the function to extract the value
	}
	
}
]
}
```

This would allow to remove the mappedValueIdCommandClasses and mappedExistingCommandClasses, you could just add a check if the valueId CC exists in the mappedNodeProps keys and if so for every definition in the array run the function to extract the value

Reply:
@robertsLando , yes, that's a good idea - much cleaner and more flexible :-).
I may have to separate the function for value mapping from the function that checks for the existence of a command class.
E.g.:
```typescript
const mappedNodeProps = {
  [CommandClasses.Battery]: [
    {
      nodeProp: 'batteryLevelMin',
      valueFn: (node, value) => { ... }
    },
    {
      nodeProp: 'powerSource',
      existsFn: (node) => { ... }
    },
  ]
}
```

Reply:
Or you can manage it by make the fn return an object, try and check what makes it more cleaner :)